keps/0023-enable-disable-features.md

```diff
@@ -0,0 +1,131 @@
+---
+kep-number: 23
+title: Enabling and Disabling features in KUDO Operators
+authors:
+  - "@zmalik"
+owners:
+  - "@zmalik"
+creation-date: 2020-01-28
+last-updated: 2020-01-28
+status: provisional
+---
+
+# Enabling and Disabling features in KUDO Operators
+
+## Summary 
+
+KUDO enables Operator developers to create Operators and develop features for them. 
+The goal of this KEP is to facilitate operator developers to enable and disable the features in a deterministic way.
+
+## Motivation
+
+We want to provide operator users a way to enable and disable features in their operator. And right now there is no operator-user friendly or operator-developer friendly way to do it. 
+
+### Goals
+
+Given a feature, `A` is composed of Kubernetes object `x`, `y` and `z`
+ 
+- create or update kubernetes objects  `x`, `y` and `z`, when feature `A` is enabled.
+- delete kubernetes objects  `x`, `y` and `z`, when feature `A` is disabled.
+
+The goal of this KEP is to provide a way to create and delete objects `x`, `y` and `z`. 
+
+### Non-Goals
+
+- complex logic like loops to trigger multiple features
+- enable/disable nested features
+- evaluate multiple conditions to enable one feature
+
+## Proposal
+
+### Using Instance Parameters
+
+In KUDO, specific plans can be triggered when updating a parameter. Plans are specifically formed by phases, steps and utimately by tasks.
+And when talking about enabling/disabling features, we are eventually talking about the `Kind` of the Task. Currently, we have `Apply`, `Delete` and `Pipe` tasks kind.
+Enabling a feature is running a task of `Apply` kind and disabling is running a task of `Delete`
+
+We propose introducing a new Task kind named `Feature`. The Feature kind handles Apply or Delete resources based on the value of a parameter in the Operator Instance.
```

Comment:
I would prefer the name FeatureToggle but no strong opinion here :)

Reply:
I like just `Toggle` if we're going with the proposed approach

Reply:
sounds good to me! if I get more +1 to that I will change to `FeatureToggle`
My only reason for using `Feature` was to keep using one-word `Task` kinds. Even if underneath implementation isn't as simple as the one word might represent

Reply:
I'd probably prefer `Toggle` as well, it describes what it does and it's a one-work task. `FeatureToggle` would be my second choice.